FreeRTOS-Plus/Demo/FreeRTOS_Plus_POSIX_with_actor_Windows_Simulator/lib/FreeRTOS-Plus-POSIX/source/FreeRTOS_POSIX_pthread.c

```diff
@@ -473,6 +481,43 @@ int pthread_join( pthread_t pthread,
     return iStatus;
 }
 
+/*-----------------------------------------------------------*/
+int pthread_cancel( pthread_t pthread )
+{
+    int iStatus = 0;
+    pthread_internal_t * pxThread = (pthread_internal_t *) pthread;
+    
+    if (pxThread == NULL || pxThread->xTaskHandle == NULL) 
```

Comment:
Similar to the comment made in ```pthread_join()```. Prefer to stick to standard. 

The alternative is to define whether caller needs to check null before calling this API.